docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,1245 @@
+  </TabItem>
+</Tabs>
+
+**Verifying Installation**
+
+Type `pact` in a terminal and try out some commands:
+
+```clojure
+$ pact
+pact> (+ 1 2)
+3
+pact> (+ "hello, " "world")
+"hello, world"
+pact> (< 0 1)
+true
+```
+
+### Create Project Structure
+
+Copy paste the instructions below to create a basic project structure:
+
+```bash
+mkdir my-pact-project && cd my-pact-project
+mkdir pact
+mkdir app
+```
+
+### Atom IDE (Optional)
+
+You can use any text editor to write Pact but if you prefer the benefits of an IDE, "language-pact" is a package for [Atom](https://atom.io) that provides syntax highlighting and linting.
+
+To install it, go to Preferences -> Packages -> Type "language-pact" and click the Install button.
+
+![atom-language-pact](/img/docs/voting-dapp/atom-language-pact.png)
+
+## Smart Contracts
+
+Our voting app should allow you to submit a vote while not allowing an address to vote more than once.
+Additionally we will use a *gas station* to fund the gas fees for interacting with our contract, meaning our users don't have to worry about paying gas.
+
+:::info
+**Gas** is the cost necessary to perform a transaction on the network. Gas is paid to miners and its price varies based on supply and demand. It's a critical piece of the puzzle, but at the same time it brings up a UX problem. Every user needs to be aware of what gas is as well as how much gas they need to pay for their transaction. This causes significant friction and a less than ideal experience.
+
+To help mitigate this problem Kadena brings an innovation to the game. Hello **gas stations**! Gas stations are a way for dApps to subsidize gas costs for their users. This means that your user doesn't need to know what gas is or how much the gas price is, which translates into a smooth experience when interacting with your dApp.
+:::
+
+### Voting
+
+Let's start to implement our voting contract.
+
+A typical developer workflow looks like this:
+
+1. Write logic in `.pact` files
+2. Write tests in `.repl` files
+3. Execute your tests in the repl
+4. Deploy to local pact server
+5. Deploy to testnet
+
+Here we will be focusing on steps 1-3.
+
+Assuming you are still in your project directory, create the `election.pact` and `election.repl` files:
+
+```clojure
+touch pact/election.pact pact/election.repl
+```
+
+`election.pact` is the contract source code while in `election.repl` we'll write tests.
+
+We will also need the `coin` contract which requires the `fungible-v2` interface for our tests. You can get the latest `coin-v3` [here](https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/v3/coin-v3.pact) and `fungible-v2` [here](https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/v2/fungible-v2.pact). Make sure to add those files in your project.
```

Comment:
Best practice dictates that these need to be placed in `pact/root/` (see structure here: https://github.com/kadena-io/marmalade/tree/main/pact/root)

Reply:
Moved them in the source code repo as well